From: Michael R Sweet <msweet@msweet.org>
Date: Sat, 6 Mar 2021 16:16:42 -0500
Subject: Make sure 'lang' is initializes.

---
 cups/testlang.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cups/testlang.c b/cups/testlang.c
index 057de84..3c83ca6 100644
--- a/cups/testlang.c
+++ b/cups/testlang.c
@@ -50,7 +50,7 @@ main(int  argc,				/* I - Number of command-line arguments */
   const char	*opt;			/* Current option */
   int		errors = 0;		/* Number of errors */
   int		dotests = 1;		/* Do standard tests? */
-  const char	*lang;			/* Single language test? */
+  const char	*lang = NULL;		/* Single language test? */
   cups_lang_t	*language = NULL;	/* Message catalog */
 
 
